docs/how-to-guides/troubleshoot.md

```diff
@@ -0,0 +1,174 @@
+# FHIR Converter Troubleshooting Guide
+ 
+## Error Response Body
+ 
+All error response bodies are returned in JSON format following the below structure:
+ 
+```
+{
+    "error": {
+        "code": <top-level error code>,
+        "message": <description of the top-level error>,
+        "innerError": {
+            "code": <error code corresponding to the inner error>,
+            "message": <message describing the inner error>
+        }
+    }
+}
+```
+ 
+The top-level error code is a string value that serves as a high-level description of the error. This value is equivalent to the value of the "x-ms-error-code" response header. Both of these fields are bound by the API contract; this means that a change to the error code that is returned for a given set of conditions will not change without an API version update. Additionally, no new top-level error codes will be added to the API without a version update. All remaining fields in the response body are *not* bound to the API contract, which means that these values may change without an API version update and are intended for end-user debugging, not to be relied upon by the client code. The top-level error code and message are guaranteed to be populated, but the InnerError will be empty for 500-level responses.
+ 
+## Debugging with application logs
+ 
+Application logs can provide detailed debugging information specific to the error. Error logs conform to the following format:
+ 
+```
+Failed conversion request with HTTP Status {HTTP status}; TOP-LEVEL-ERROR: {error code} - {top-level error message}.; INNER ERROR: Code: {inner error code}, Message: {inner error code}, Exception: {the outer exception that causes the Inner Error}; InnerException: {the inner exception of the Inner Error} - Code: {error code of the inner exception}, Message: {inner exception message}.
+```
+ 
+The top-level error information in the error log aligns with the top-level error fields in the response body. If the inner error details are present in the response body, they align with the inner error details of the log message, though the log message will likely provide more details than the response body. See the [Monitoring](monitoring.md) document for more information on accessing these logs and sample queries that may be helpful for debugging.
+ 
+## Top-Level Error Codes
```

Comment:
Suggestion: I think a tabular format might make this more user friendly. Especially since the columns are the same, i.e., Top level, inner error, root cause, troubleshooting